### DISCUSSION TITLE
q-select not showing option list on first try after page load

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#10131] [General - Components / Directives / etc] [answered]
Hi,

I'm using q-select to filter values with an axios call.
I used the update function as showed in the docs, everything works fine but option list is not showing on first try after page load.
It triggers the axios call, wich is responding normally but q-select doesn't show the options list.
After this issue, if I click in the select to hide then click again to display the list it's shown normally and every next try will work fine.

Any idea ?

**Comment by metalsadman** [comment, marked as answer]
need codepen, my guess, see the api cards for props like the `map-options` and `emit-value`. etc...

**Reply by ghost** [reply]
Hi

Thanks for your advice, we already used these props, we just found the solution, we've had to make the filter function async/await for it to work.
